src/nameservice/WFServiceGovernance.cc

```diff
@@ -69,13 +69,13 @@ class WFSelectorFailTask : public WFRouterTask
 
 static void copy_host_port(ParsedURI& uri, const EndpointAddress *addr)
 {
-	if (addr->host != uri.host)
```

Comment:
I strongly suggest adding protective judgment when comparing std::string with char *(which may be nullptr), because `class WFServiceGovernance` 's requests are not necessarily come from `class WFComplexClientTask`(which set port to uri), as long as it is nullable in any case, the code may core in an uncertain future.

Reply:
Thank you for your carefulness. Already fixed in https://github.com/sogou/workflow/pull/777.

Reply:
Make life more beautiful 😁